### PR TITLE
Add an only-suspicious-option to filter strings

### PR DESCRIPTION
**Pull request by Ana06**
Add an `only-suspicious-option` which filters potential domains, urls, files (without including the Windows ones) and registers.

It may be possible to improve the regular expressions. I tried to keep the regular expressions simple (i.e. it returns strings similar to IPs that are not IPs such as `299.0.0.0`) to avoid making the code difficult to read/modify. I also made them strict (i.e. it doesn't return files with short names) to avoid returning too many strings. We may miss some strings, but we may also get the most important strings really fast. I tried it in a few files and it seems to work quite well. :tada: The list of Windows files is also not complete. I am not sure if we can get a complete list of files somewhere. :thinking: 

I need to write a test and documentation for the new option, but I would like to receive feedback about the feature before doing so.

Example:

```
> python floss/main.py --only-suspicious-strings ana_malware.exe
FLOSS static ASCII strings
anaisgreat.dll
\system32\anaisgreat.exe
http://anaisgreat.com/file.exe
192.178.63.1

FLOSS static UTF-16 strings


FLOSS decoded 2 strings
ana06.exe

FLOSS extracted 2 stackstrings
ana06.exe

```
Closes https://github.com/fireeye/flare-floss/issues/304

### CODE REVIEW

File: floss/main.py [hunk 1]
```diff
@@ -483,7 +496,7 @@ def print_identification_results(sample_file_path, decoder_results):
             headers=["address", "score"]))
 
 
-def print_decoding_results(decoded_strings, group_functions, quiet=False, expert=False):
+def print_decoding_results(decoded_strings, group_functions, quiet=False, expert=False, only_suspicious_strings=False):
```
Comment (williballenthin) [inline floss/main.py]: these prototypes are getting a bit out of control (not an issue specific to this PR). lots of options and lots of possible combinations.

@mr-tz:

the code can be made more succinct if we use a config object (e.g. a dict of options); however, this doesn't mitigate the number of things to track.

we may want to spend a little time thinking about the different classes of output options and see if we can refactor here. the goal to make things concise and easy to understand.

for example, there may be multiple types of filter that we do (suspicious, static strings, unique, etc.) and maybe we could configure and implement these in a similar way. probably just needs some time dedicated to this.

File: floss/main.py [hunk 2]
```diff
@@ -42,6 +43,16 @@
 
 MIN_STRING_LENGTH_DEFAULT = 4
 
+WINDOWS_FILES = ['KERNEL32.DLL', 'GDI32.DLL', 'USER32.DLL', 'COMCTL32.DLL',
+                 'COMDLG32.DLL', 'WS2_32.DLL', 'ADVAPI32.DLL', 'NETAPI32.DLL',
+                 'OLE32.DLL', 'SHELL32.DLL', 'WUSER32.DLL', 'PSAPI.DLL',
+                 'MSVCRT.DLL', 'CRTDLL.DLL', 'MSCOREE.DLL']
```
Comment (williballenthin) [inline floss/main.py]: this is a good initial list. how did you pick these? should we add a comment saying "These are referenced in PMA" or similar?

File: floss/main.py [hunk 3]
```diff
@@ -42,6 +43,16 @@
...
+
+IS_IP = "^([12]{,1}\d{1,2}\.){3}[12]{,1}\d{1,2}"
+IS_DOMAIN_FILE = "^[\\\\\w\/:@\.-]{5,256}\.[a-zA-Z0-9]{1,4}"
```
Comment (williballenthin) [inline floss/main.py]: can you add comments that describe:

  1. how these work?
  2. examples of what they should match?

the IP regex is neat - it took me a minute or two to figure out how it works.

File: floss/main.py [hunk 4]
```diff
@@ -42,6 +43,16 @@
...
+IS_REGISTER = "^(HKEY_LOCAL_MACHINE\\\\|HKLM\\\\).+$"
```
Comment (williballenthin) [inline floss/main.py]: should be `IS_REGISTRY_HIVE`

also include:
  - HKCU
  - HKU

Comment (williballenthin) [reply floss/main.py]: should anchoring at the start/end of the strings be consistent across all these patterns?

File: floss/main.py (hunk 2 above)
Comment (williballenthin) [reply floss/main.py]: this should be a Set rather than List

**Comment by williballenthin** [comment]
I think the implementation here is fine. I'm not 100% convinced on the feature, but I look forward to more discussion on it. I'll add my thoughts to #304

File: floss/main.py (hunk 2 above)
Comment (Ana06) [reply floss/main.py]: @williballenthin 
> this is a good initial list. how did you pick these? should we add a comment saying "These are referenced in PMA" or similar?

Wikipedia + some that tend to appear in some files I tried this with. As said in the PR description:
> The list of Windows files is also not complete. I am not sure if we can get a complete list of files somewhere.

**Comment by Ana06** [comment]
I haven't update the PR, because it seems it is not clear if this is the best solution. In case we decide this is wanted, I will update the PR. :wink: 

I have also realized of a few cases that are not cover and that would be really easy to add. For example, port numbers are not currently included.

**Comment by mr-tz** [comment]
I propose to close this and come back to it once we've addressed the Python 3 cutover and #328

**Pull request closed without merging**